### PR TITLE
[Runtime/Lua]  Support multiple results in executeFunctionWithLuaBackend

### PR DESCRIPTION
**Pull request by jhalakpatel**

### CODE REVIEW

**Comment by christopherbate** [comment]
IIUC, we should be able to add tests here?

**Comment by jhalakpatel** [comment]
> IIUC, we should be able to add tests here?

can you point me to an example of such tests? IIUC, we need complete IR translation to work to reach a point where function actually returns a memref value to actually the test the code being added. I will be adding several integration tests in the later PR once core changes are merged.

File: mlir-tensorrt/executor/include/mlir-executor/Runtime/Backend/Lua/LuaRuntime.h [hunk 1]
```diff
@@ -93,12 +93,14 @@ StatusOr<int64_t> runExecutorExecutable(
...
-/// output (destination args). Returns any results.
+/// output (destination args). Returns optional results supporting both DPS and
+/// non-DPS style calling convention.
 StatusOr<llvm::SmallVector<std::unique_ptr<RuntimeValue>>>
 executeFunctionWithLuaBackend(LuaRuntimeSession &session, std::string_view name,
                               llvm::ArrayRef<RuntimeValue *> inputArgs,
                               llvm::ArrayRef<RuntimeValue *> outputArgs,
-                              std::optional<CudaStream> stream = {});
+                              std::optional<CudaStream> stream = {},
+                              std::optional<RuntimeClient *> client = {});
```
Comment (christopherbate) [inline mlir-tensorrt/executor/include/mlir-executor/Runtime/Backend/Lua/LuaRuntime.h:103]: You'll have to explain what this argument is for/when to provide it in the documentation comment.

File: mlir-tensorrt/executor/lib/Runtime/Backend/Lua/LuaRuntime.cpp [hunk 2]
```diff
@@ -502,12 +502,138 @@ static Status validateArgsTypesAgainstFuncArgs(const RuntimeValue *runArg,
   return getOkStatus();
 }
 
+[[maybe_unused]] static constexpr int MEMREF_FIXED_FIELDS =
```
Comment (christopherbate) [inline mlir-tensorrt/executor/lib/Runtime/Backend/Lua/LuaRuntime.cpp:505]: For constant integers, our convention is to use `int64_t` and naming as follows:

```
static constexpr int64_t kMemRefFixedFields = ...
```

Comment (jhalakpatel) [reply mlir-tensorrt/executor/lib/Runtime/Backend/Lua/LuaRuntime.cpp:505]: This is not present in the final code anymore.

**Comment by shelkesagar29** [comment]
For failing test, 
1. Lets add new test to a folder named `Lua`.
2. `Lua` folder should be compiled only if `MLIR_EXECUTOR_ENABLE_GPU_INTEGRATION_TESTS ` is ON
```
if(MLIR_EXECUTOR_ENABLE_GPU_INTEGRATION_TESTS)
    add_subdirectory(Lua)
endif()
```

INT4 tests don't need GPU

**Comment by shelkesagar29** [comment]
Also, commit message doesn't follow canonical format. Can you please fix it?

**Review by shelkesagar29 — APPROVED** [review]

**Pull request merged**